src/pages/node-express/lesson12-a1.mdx

```diff
@@ -0,0 +1,473 @@
+This time, using server-side rendering with the `EJS` templating language. If you haven't started your final
+project, you can use this project as the basis, using server-side rendering instead of the
+front-end + back-end model. Of course, to satisfy the requirements of the rubric, you would
+need to modify the schema to manage objects that are different from the jobs model.
+
+## Creating the Repository
+
+There is no starter repository for this project, so you will need to create one, with the following
+steps. First, create a `jobs-ejs` folder/directory on your computer (not in the tree of a previous project),
+and `cd` (change directories) into it. **Make sure that it is not in the tree of a previous
+project, that is, running `git status` _should_ return an error after you create the directory**. Next,
+create the `.gitignore` file and the `.env` file. The `.gitignore` file is critical to make sure
+your Mongo credentials and the Node libraries are not stored in Github. It should have these lines:
+
+```
+/node_modules
+.env
+.DS_Store
+```
+
+You can just copy the `.gitignore` file and the `.env` file from the `06-jobs-api` directory.  
+You won't use the JWT values from `.env`, so you can delete those. The
+`.env` file is needed for the Mongo credential, and eventually for the session secret.
+
+Next, run `git init` to make the directory a git repository. Then, log into Github and create
+a new repository called `jobs-ejs`. You create a new repository using the + button in the
+upper right of your Github screen. You do not use a template, the repository should be
+public, and you do not create a `README` or a `.gitignore`.
+
+![create new repo button](/create-new-repo.png)
+
+Once the repository has been created, you need to associate the Github repository with the repository on your laptop.
+You will see the following screen:
+
+![The github page for a newly created repository](/public/github-new.png)
+
+Click on the clipboard icon next to the URL to copy it. Then, in your laptop session for
+the jobs-ejs repository, type the following, where the URL is the one you copied to your
+clipboard
+
+```
+git remote add origin <paste URL here>
+git add -A
+git commit -m "first commit"
+git push origin main
+```
+
+The local repository is now associated with your Github repository. There isn't much
+in it yet, just the `.gitignore`. Now create the `lesson12` branch, where you will do your
+work.
+
+## Components and Directories
+
+You need to initialize NPM for your repository. So, do an `npm init`. You can take all
+the defaults when prompted. This creates the `package.json`, and enables the installation
+of npm packages. You need to add `scripts` for dev and start to the `package.json` so that
+you can do `npm run` or `npm run dev`, where the `dev` script runs app.js using `nodemon` and the plain `npm run` runs it using `node`.
+You need to install the following packages:
+
+```
+bcryptjs
+connect-flash
+connect-mongodb-session
```

Comment:
This is used later in the assignment (#Sessions section) but was not included in packages needed to be instaleld for this assignment